tripal_chado/src/Plugin/TripalImporter/FASTAImporter.php

```diff
@@ -0,0 +1,1148 @@
+        //   '!name' => $name,
+        //   '!uname' => $uname,
+        // ], TRIPAL_ERROR);
+        $this->logger->error("Failed to retrieve newly inserted feature '$name ($uname)'");
+        return 0;
+      }
+
+      // Add the residues for this feature
+      $this->loadFastaResidues($fh, $feature->feature_id, $seq_start, $seq_end);
+    }
+
+    // if we don't have a feature and the user wants to do an update then fail
+    if (!isset($feature) and (strcmp($method, 'Update only') == 0 or strcmp($method, 'Insert and update') == 0)) {
+      // $this->logMessage("Failed to find feature '!name' ('!uname') while matching on " . drupal_strtolower($match_type) . ".",
+      //   ['!name' => $name, '!uname' => $uname], TRIPAL_ERROR);
+      $this->logger->error("Failed to find feature '$name' ('$uname') while matching on " . mb_strtolower($match_type) . ".");
+      return 0;
+    }
+
+    // if we do have a feature and this is an update then proceed with the update
+    if (isset($feature) and !$inserted and (strcmp($method, 'Update only') == 0 or strcmp($method, 'Insert and update') == 0)) {
+
+      // if the user wants to match on the Name field
+      if (strcmp($match_type, 'Name') == 0) {
+
+        // if we're matching on the name but do not have a unique name then we
+        // don't want to update the uniquename.
+        $values = [];
+        if ($uname) {
+
+          // First check to make sure that by changing the unique name of this
+          // feature that we won't conflict with another existing feature of
+          // the same name
+          $values = [
+            'organism_id' => $organism_id,
+            'uniquename' => $uname,
+            'type_id' => $cvterm->cvterm_id,
+          ];
+          $results = chado_select_record('feature', ['feature_id'], $values);
+          if (count($results) > 0) {
+            // $this->logMessage("Cannot update the feature '!name' with a uniquename of '!uname' and type of '!type' as it " .
+            //   "conflicts with an existing feature with the same uniquename and type.",
+            //   [
+            //     '!name' => $name,
+            //     '!uname' => $uname,
+            //     '!type' => $cvterm->name,
+            //   ], TRIPAL_ERROR);
+            $this->logger->error("Cannot update the feature '$name' with a uniquename of '$uname' and type of '" . $cvterm->name . "' as it " .
+            "conflicts with an existing feature with the same uniquename and type.");
+            return 0;
+          }
+
+          // the changes to the uniquename don't conflict so proceed with the update
+          $values = ['uniquename' => $uname];
+          $match = [
+            'name' => $name,
+            'organism_id' => $organism_id,
+            'type_id' => $cvterm->cvterm_id,
+          ];
+
+          // perform the update
+          $success = chado_update_record('feature', $match, $values);
```

Comment:
`chado_update_record` is deprecated. We should only be using Tripal DBX for all chado related queries.